server/controllers/taskController.js

```diff
@@ -0,0 +1,102 @@
+import asyncHandler from "express-async-handler";
+import Workspace from "../models/workspaceModel.js";
+
+// @desc    Create a new task
+// @route   POST /api/workspaces/:workspaceId/boards/:boardId/columns/:columnId/tasks
+// @access  Private
+const createTask = asyncHandler(async (request, response) => {
+  const { name, description } = request.body;
+  const currentUserId = request.user._id;
+  const workspaceId = request.params.workspaceId;
+  const boardId = request.params.boardId;
+  const columnId = request.params.columnId;
+
+  // Check if the workspace exists
+  let workspace;
+  try {
+    workspace = await Workspace.findById(workspaceId);
+  } catch (error) {
+    response.status(404);
+    throw new Error("Workspace not found");
+  }
+  // Check against old deleted workspace IDs
+  if (!workspace) {
+    response.status(404);
+    throw new Error("Workspace not found");
+  }
+
+  // Check if the user has access to the workspace
+  if (
+    workspace.creator.toString() !== currentUserId.toString() &&
+    !workspace.members.includes(currentUserId)
+  ) {
+    response.status(403);
+    throw new Error("Unauthorized access to this workspace");
+  }
+
+  // Find the board
+  const board = workspace.boards.id(boardId);
+
+  // Check if the board exists
+  if (!board) {
+    response.status(404);
+    throw new Error("Board not found");
+  }
+
+  // Find the column
+  const column = board.columns.id(columnId);
+
+  // Check if the column exists
+  if (!column) {
+    response.status(404);
+    throw new Error("Column not found");
+  }
+
+  // Check if a task with the same name already exists in the column
+  const taskExists = column.tasks.find((task) => task.name === name);
```

Comment:
I think it should be okay to have a task with the same name, so we can remove this logic